code/__HELPERS/radio.dm

```diff
@@ -8,12 +8,26 @@
 	if(!(. % 2)) // Ensure the last digit is an odd number
 		. += 1
 
-// Format frequency by moving the decimal.
+/// Format frequency by moving the decimal.
 /proc/format_frequency(frequency)
 	frequency = text2num(frequency)
 	return "[round(frequency / 10)].[frequency % 10]"
 
-//Opposite of format, returns as a number
+///Opposite of format, returns as a number
 /proc/unformat_frequency(frequency)
 	frequency = text2num(frequency)
 	return frequency * 10
+
+///returns a random unused frequency between MIN_FREE_FREQ & MAX_FREE_FREQ if free = TRUE, and MIN_FREQ & MAX_FREQ if FALSE
+/proc/return_unused_frequency(free = FALSE)
+	var/start = free ? MIN_FREE_FREQ : MIN_FREQ
+	var/end = free ? MAX_FREE_FREQ : MAX_FREQ
+
+	var/freq_to_check = rand(start, end)
+	freq_to_check += !(freq_to_check % 2) //make sure freq_to_check is odd
```

Comment:
This isn't readable, and it abuses that booleans aren't real. Just make an if statement.

Reply:
done